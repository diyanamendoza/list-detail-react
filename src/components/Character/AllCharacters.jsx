import { Link } from 'react-router-dom'


export default function AllCharacters({ characters }) {
    return (
        <>
            <ul aria-label="characters">
                {characters.map((character) => (
                <li key={character._id+character.name}>
                    <Link to={`/characters/${character._id}`}>
                    <figure>
                        <img
                        src={character.photoUrl}
                        alt={character.name}
                        />
                        <figcaption>
                        {character.name}
                        </figcaption>
                    </figure>
                    </Link>
                </li>
                ))}
            </ul>
      </>
    )
  }
  