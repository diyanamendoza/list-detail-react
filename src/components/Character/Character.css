@import url('https://fonts.googleapis.com/css2?family=Source+Code+Pro:wght@200&display=swap');


img {
    border-radius: 1em;
}

figure {
    padding: 10px;
    color: white;
    font-family: 'Source Code Pro', monospace;

}