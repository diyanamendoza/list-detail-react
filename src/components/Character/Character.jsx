import './Character.css'

export default function Character({character}) {
    return (
        <figure>
        <img src={character.photoUrl} alt={character.name} />
        <figcaption>
          <h2>{character.name}</h2>
          <p>FAMBAM: {character.affiliation}</p>
          <p>LOVAS: {character.allies}</p>
          <p>HATERS: {character.enemies}</p>
        </figcaption>
      </figure>
    )
}
