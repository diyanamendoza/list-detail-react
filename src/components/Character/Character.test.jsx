import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Character from './Character';

it('should display the provided character', () => {
  const { container } = render(
    <MemoryRouter>
      <Character
        character={{
          photoUrl: 'https://placebear.com/200/300',
          name: 'Ted',
          affiliation: 'bear nation',
          allies: 'bees',
          enemies: 'humans',
        }}
      />
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
  screen.getByText(/Ted/i)
});
