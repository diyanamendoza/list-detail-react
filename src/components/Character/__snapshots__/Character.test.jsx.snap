// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display the provided character 1`] = `
<div>
  <figure>
    <img
      alt="Ted"
      src="https://placebear.com/200/300"
    />
    <figcaption>
      <h2>
        Ted
      </h2>
      <p>
        FAMBAM: 
        bear nation
      </p>
      <p>
        LOVAS: 
        bees
      </p>
      <p>
        HATERS: 
        humans
      </p>
    </figcaption>
  </figure>
</div>
`;
